src/fal/dag.py

```diff
@@ -0,0 +1,129 @@
+from dataclasses import dataclass, field
+from os import path
+from typing import List,TypeVar, Dict
+from faldbt.project import DbtModel 
+from pathlib import Path
+import re
+
+T = TypeVar("T", bound="FalScript")
+
+@dataclass(frozen=True)
+class FalScript:
+    model: DbtModel
+    path: str
+    dependencies: List[T] = field(default_factory=list)
+
+    def __hash__(self):
+        return self.path.__hash__()
+
+    def __repr__(self):
+        return "FalScript(" + self.path.__str__() + ")"
+
+    def parse(self) -> List[str]:
+        """
+        Parses the python file to identify any dependencies
+        """
+        with open(self.path) as file:
+            lines = file.readlines()
+            ## TODO: do this search in parallel
+            dependency_names = []
+            for line in lines:
```

Comment:
Seems to be very easy with the `with` approach: https://docs.python.org/3/library/multiprocessing.html#introduction

Reply:
if its ok with you i am going to delay this as a future task..